src/main/java/org/spongepowered/common/event/manager/SpongeEventManager.java

```diff
@@ -318,7 +319,15 @@ public <E extends Event> EventManager registerListener(final EventListenerRegist
 
     @Override
     public SpongeEventManager registerListeners(final PluginContainer plugin, final Object listener) {
-        this.registerListener(plugin, listener);
+        this.registerListener(plugin, listener, null);
+        return this;
+    }
+
+    @Override
+    public SpongeEventManager registerListeners(final PluginContainer plugin, final Object listener,
+                                                final MethodHandles.Lookup lookup) {
+        Objects.requireNonNull(lookup, "lookup");
```

Comment:
Is it the expectation that the plugin providing the lookup will have already prepared private lookups in it?

https://github.com/SpongePowered/Sponge/blob/1434c20ed1455db0a0df9039c96401c8a425947b/src/main/java/org/spongepowered/common/event/manager/ClassEventListenerFactory.java#L90

Like there where, since we create the lookup, we call that on that internal lookup.